### ISSUE TITLE
make "Reduced by ignore patterns" label act like "editIgnoresButton"

### ISSUE THREAD

**Issue by katrinleinweber** [syncthing/syncthing#4293]
Hello!

Sometimes, when I see a folder's `Local State` hintin at ignores, I ask myself "Which rules?" and whether I should review them? So, I wondered whether [that label](https://github.com/syncthing/syncthing/blob/master/gui/default/index.html#L335) could be upgraded with [the `Edit | Ignore Patterns` button's functionality](https://github.com/syncthing/syncthing/blob/master/gui/default/syncthing/folder/editFolderModalView.html#L196) just to shave off that one click from the current click path.

I've not done a lot of publishable HTML or JavaScript, but this seems to me like an easy hack. Would a GUI tweak that hyperlinks the label, or introduces a small button besides it be welcome? And safe to PR against your test suite?

**Comment by AudriusButkevicius** [comment]
So personally I'd be against it. There is already the way to do what you want, and I am fine with having one way only to do that.

**Comment by katrinleinweber** [comment]
What is the way to do it? On my 1280x800px screen, there's a good chance that `Reduced by ignore patterns` is "above the fold", so to speak, while the same folder's `Edit` button is "below." That's one scroll to do. The folder's modal is always larger than the screen is high, so that's always a 2nd scroll. Both could be shaved off by my idea, plus one of the button clicks.

Alternatively, could the folder panels be upgraded to scroll the view down as much as needed to keep the next folder's panel-heading on-screen?

**Comment by AudriusButkevicius** [comment]
You should zoom out. Yes, things don't fit in your screen,  and you have to click and  scroll but that's due to the size of your screen. 

Things won't fit if you have many folders too, but that doesn't means we should pepper clickable things across the screen because of that. 

Your idea brings incosistency, as you can't always click that label as its not always visible. I suspect we'll have people turning up saying "how do I open ignores modal for new folders, the 'reduced by' message is not visible"

**Comment by katrinleinweber** [comment]
People who notice my suggested shortcut, would do so _only_ if they first created ignore rules by the standard, long way. That would of course remain in place for all who don't care about a shortcut.
But for those who do notice the shortcut, it could be a nice touch. Thus, I find the risk of inconsistency or confusion very low, because the two cases remain clearly distinct. Just one of them get's a shortcut.

About zooming: Do you think it scales better than a shortcut?

When I look at the [stats `Number of Folders`](https://data.syncthing.net/), I gather that power-users with more folders (more `Edit` buttons slipping below the screen edge) would need to zoom out farther and farther, and that is limited by font size, isn't it?

My example is a bit simple maybe, but: At 100% in Chrome on my 800p screen, only the 1st to 4th folder panel is fully visible if expanded. At the same time, the `reduced` label is visible down to the 7th. That's 75% more folders whose ignore rules would be more conveniently accessible for users above some %ile between the 50th and 95th (maybe 60th %ile). Yeah, it's not a big leap forward, but then again: who doesn't need it can ignore it.

Also, as I said initially, one alternative GUI idea would be to make an expanding folder panel keep itself above the fold entirely.

**Comment by AudriusButkevicius** [comment]
So anyways, I am against this, as we are solving your specific small screen problem. I guess it's your opinion against mine, so lets see if someone else from the team/users has anything to say about this. If I am the only one who thinks this is cluttering UI for everyone to help people who are not bothered to scroll on small screens, then I'll shut up.

**Comment by AudriusButkevicius** [comment]
Syncthings UI is a train wreck anyway by the way, so less clutter the better imho.

**Comment by katrinleinweber** [comment]
You're also solving my specific syncing needs that I can't be bothered to solve with thumb-drives ;-p

**Comment by calmh** [comment]
I don't think the Syncthing UI is a train wreck. Than again I would say that, I'm responsible for most of it. What I do think is that the dialogs are somewhat atrocious. There will always be a certain amount of scrolling on small screens, but we don't necessarily need to make that worse by having shaded background and margins and rounded corners take up literally half the screen. I think we can improve things in small steps by adding more full page dialogs and potentially top level tabs, without having to redesign the whole UI. That's something of a side track though. But it goes against the broken window "the UI is anyway trash so no point in improving it" feeling.

Making this a "Reduced by <a>ignore patterns</a>" thing with the ignore patterns being a link to the editor doesn't seem like it adds significant amount of clutter, to me, and seems like it could be a useful enhancement. The ignore patterns are hidden a couple of levels into the edit dialogs after all.

It's somewhat analogous to how we provide links for errors and in progress items from the folder status screen.

Anyway, musings from the beach. A non-horrible pull request would likely be merged by me. :)

**Comment by AudriusButkevicius** [comment]
Well, it was not a train wreck when you started started, but then we just kept bolting stuff on top as modals that pop up when you click weird things, so I guess this would just be yet another weird thing you can click on that produces magic.

**Comment by calmh** [comment]
Fair enough. I think the magic factor is low enough that this is not a WTF, regardless. But I agree that the GUI is an organic accretion of stuff, and about as palatable as that sounds. It would benefit from a rethink. But we can accrete more stuff in the meantime. ;)

**Comment by calmh** [comment]
> Also, as I said initially, one alternative GUI idea would be to make an expanding folder panel keep itself above the fold entirely.

I'd be perfectly fine with the screen scrolling down to actually show the whole section when a folder or device is expanded, provided the folder/device header isn't scrolled off screen.

**Comment by katrinleinweber** [comment]
Yes, "keep itself" was meant as "scroll the view in that moment", not "stick to anything".
I'll work on the shortcut on the weekend.

**Comment by katrinleinweber** [comment]
The above would not make a good PR :-/ I got it to work on the 2nd click, after one has opened the ignore patterns the usual way once.

When adding copying the button's parameters `ng-click="editingExisting ? editIgnores() : editIgnoresOnAddingFolder()" ng-disabled="folderEditor.$invalid"`, and when adding `folder.id` in either or both function calls, a click on the label produces the error:

```
TypeError: Cannot read property 'path' of undefined
    at Scope.$scope.editIgnoresOnAddingFolder (syncthingController.js:1664)
    at angular.js:10223
    at extend.constant (angular.js:9973)
    at angular.js:17859
    at Scope.$eval (angular.js:11974)
    at Scope.$apply (angular.js:12074)
    at HTMLAnchorElement.<anonymous> (angular.js:17858)
    at HTMLAnchorElement.dispatch (jquery-2.2.2.js:4755)
    at HTMLAnchorElement.elemData.handle (jquery-2.2.2.js:4567)
 ```

This is as far as I got, after a HTML & CSS tutorial. Those didn't prepare me for JavaScript errors ;-D   I'm not sure how to puzzle together. 
 
 - this `path` to fix this error, and 
 - the CSS class to make harmonise the now hyper-linked label's style with the other hyper-links,
 
but I can't invest more time into JS in the foreseeable future, I'm sorry. Anyone who wants to try further may of course build on the above :-) Otherwise, feel free to close this for now.
